src/hooks/useGuestStatus.ts

```diff
@@ -0,0 +1,18 @@
+import { useState, useEffect } from 'react';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { LOGGED_AS_GUEST } from '@src/constants/storage';
+
+export const useGuestStatus = () => {
```

Comment:
This won't work, since this hook will have no idea that value of LOGGED_AS_GUEST has changed. This requires some global state management. Do you have zustand in this project?